haystack/utils/type_serialization.py

```diff
@@ -20,54 +21,66 @@ def serialize_type(target: Any) -> str:
     Serializes a type or an instance to its string representation, including the module name.
 
     This function handles types, instances of types, and special typing objects.
-    It assumes that non-typing objects will have a '__name__' attribute and raises
-    an error if a type cannot be serialized.
+    It assumes that non-typing objects will have a '__name__' attribute.
 
     :param target:
         The object to serialize, can be an instance or a type.
     :return:
         The string representation of the type.
-    :raises ValueError:
-        If the type cannot be serialized.
     """
-    # If the target is a string and contains a dot, treat it as an already serialized type
-    if isinstance(target, str) and "." in target:
-        return target
```

Comment:
is this part covered now?

Reply:
I can check, but all tests were passing before and after the changes I made which tells me it either is or no test for this was made.

Reply:
Yes it does work, added a new unit test for the behavior